### DISCUSSION TITLE
all the message's height has increased by 1

### DISCUSSION THREAD

**Discussion by zenghao306** [filecoin-project/lily#335] [Q&A] [answered]
At  the table [receipts],all the message's height has increased by 1.for example,the message [bafy2bzacea757tsgvogg3ngqui6tdwlx54u3y4qupxg42rzxljhldksfxs4us]'s really height is 335529,but recorded 335530 in the talbe.

**Comment by frrist** [comment, marked as answer]
This is expected and here is why:

The message receipts in a block header are the result of executing messages in the block's parents -- this is why the receipt field in a block header is called [`ParentMessageReceipts`](https://github.com/filecoin-project/lotus/blob/6f593d53d0c08f4728a0b1217bdabcae52a551d0/chain/types/blockheader.go#L68)

> The Filecoin VM interpreter orchestrates the execution of messages from a tipset on that tipset’s parent state, producing a new state and a sequence of message receipts. The CIDs of this new state and of the receipt collection are included in blocks from the **subsequent epoch**, which must agree about those CIDs in order to form a new tipset.

- [source filecoin spec](https://spec.filecoin.io/#section-systems.filecoin_vm.interpreter)
